plugins/verification/src/main/java/org/qbicc/plugin/verification/MemberResolvingBasicBlockBuilder.java

```diff
@@ -206,13 +207,8 @@ public Value checkcast(final Value value, final TypeDescriptor desc) {
         } else if (value instanceof UndefinedLiteral) {
             // it may be something we can't really cast.
             return ctxt.getLiteralFactory().undefinedLiteralOfType(castType);
-        } else if (value.getType() instanceof PointerType pt && castType.equals(pt.getPointeeType())) {
-            /* Pointer<x> -> x */
-            if (castType instanceof CompoundType) {
-                return super.deref(value);
-            } else {
-                return super.load(super.pointerHandle(value), MemoryAtomicityMode.UNORDERED);
-            }
+        } else if (value instanceof Deref) {
+            return value;
```

Comment:
This should be OK, but once this is all functional we might want to do a more complex validation between `value.getType()` vs `castType` to ensure that the input is assignable to the output.